client/trino-client/src/main/java/io/trino/client/spooling/InlineSegmentIterator.java

```diff
@@ -63,11 +64,21 @@ protected List<Object> computeNext()
         verify(iterator.hasNext(), "Iterator should have more rows, current: %s, count: %s", currentRow, rowsCount);
         List<Object> rows = iterator.next();
         if (currentRow == this.rowsCount) {
-            iterator = null;
+            unload();
         }
         return rows;
     }
 
+    private void unload()
```

Comment:
why do you need a new method. Why not just close?

Reply:
For consistency with SpooledSegmentIterator